pytensor/tensor/rewriting/linalg.py

```diff
@@ -539,3 +539,21 @@ def svd_uv_merge(fgraph, node):
                     or len(fgraph.clients[cl.outputs[2]]) > 0
                 ):
                     return [cl.outputs[1]]
+
+
+@register_canonicalize
+@register_stabilize
+@node_rewriter([Blockwise])
+def rewrite_inv_inv(fgraph, node):
+    valid_inverses = (MatrixInverse, MatrixPinv)
+    if not (isinstance(node.op.core_op, valid_inverses)):
+        return None
+
+    potential_inner_inv = node.inputs[0].owner
+    if not (
+        isinstance(potential_inner_inv.op, Blockwise)
```

Comment:
You're looking for a nested function, like `f(g(x))`, so it makes sense to check for `node.inputs[0].owner`, which would be `f`. But if we just have a single function `g(x)`, this `inputs[0].owner` will be `None` and the next line will error. You need to check that it is not `None` as the first step of this if switch.